src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs

```diff
@@ -64,7 +65,12 @@ public AdvancedOptionPageControl(OptionStore optionStore, IComponentModel compon
             BindToOption(InsertSlashSlashAtTheStartOfNewLinesWhenWritingSingleLineComments, SplitStringLiteralOptions.Enabled, LanguageNames.CSharp);
             BindToOption(InsertAsteriskAtTheStartOfNewLinesWhenWritingBlockComments, FeatureOnOffOptions.AutoInsertBlockCommentStartString, LanguageNames.CSharp);
 
-            BindToOption(DisplayInlineParameterNameHints, FeatureOnOffOptions.InlineParameterNameHints, LanguageNames.CSharp);
+            BindToOption(DisplayInlineParameterNameHints, InlineHintsOptions.EnabledForParameters, LanguageNames.CSharp);
+            BindToOption(ShowHintsForLiterals, InlineHintsOptions.ForLiteralParameters, LanguageNames.CSharp);
+            BindToOption(ShowHintsForNewExpressions, InlineHintsOptions.ForObjectCreationParameters, LanguageNames.CSharp);
+            BindToOption(ShowHintsForEverythingElse, InlineHintsOptions.ForOtherParameters, LanguageNames.CSharp);
+            UpdateInlineHintsOptions();
```

Comment:
Does UpdateInlineHintsOptions need to be called here, or does the binding cause the Checked/Unchecked property to change? If this is necessary then this may need to go down in to the OnLoad a bit further down for the same reason as the other code there. The dialog pane is reused between invocations, and if the sequence of events is:

1. Open is enabled or disabled locally.
2. Dialog is closed.
3. Option is changed via setting sync from another machine.
4. Dialog is opened again

This code isn't running during the second reopen, whereas the code in OnLoad will. But if Checked/Unchecked is raised on binding changes, then maybe this doesn't matter at all.

Reply:
i'll put it into OnLoad.